### PR TITLE
Auto-scale + fixes on the Place parameter.

### PR DESCRIPTION
**Pull request by FunkySayu**

### CODE REVIEW

File: computation/imagefetcher/imagefetcher/fetcher.py [hunk 1]
```diff
@@ -203,17 +203,51 @@ def _GeometryToRectangle(geometry):
...
+            return x_max - x_min + y_max - y_min
+
+        max_distance, max_bounds = 0, None
+        for parts in geo_json['coordinates']:
+            bounds = get_rectangle_bounds(parts)
+            bounds_distance = distance(*bounds)
+            if bounds_distance > max_distance:
+                max_distance, max_bounds = bounds_distance, bounds
+
+        print(max_bounds)
```
Comment (FunkySayu) [inline computation/imagefetcher/imagefetcher/fetcher.py]: Remove this

File: computation/imagefetcher/imagefetcher/utils.py [hunk 2]
```diff
@@ -96,6 +96,23 @@ def get_geometry(get_parameters):
...
+        rectangle: Rectangle generated from the client query.
+    Returns:
+        A 'best scale' based on the rectangle values.
+    """
+    # Get the rectangle bounds
+    x_0, y_0 = rectangle.toGeoJSON()['coordinates'][0][0]
+    x_1, y_1 = rectangle.toGeoJSON()['coordinates'][0][2]
+
+    scale = int((abs(x_0 - x_1) + abs(y_0 - y_1)) * 80)
+    print(scale)
```
Comment (FunkySayu) [inline computation/imagefetcher/imagefetcher/utils.py]: Remove  this

File: computation/imagefetcher/imagefetcher/app.py [hunk 3]
```diff
@@ -80,7 +81,7 @@ def rgb_handler(date, polygon, place, country, city, scale, delta):
             City to visualize. Required, or country/polygon must be specified.
         scale (float):
             Precision of the picture. Unit is meter per pixels so lower is
-            better.
+            better. Attempts to automatically generates it if not specified.
```
Comment (ArchangelX360) [inline computation/imagefetcher/imagefetcher/app.py]: to ... generate without an 's'

File: computation/imagefetcher/imagefetcher/utils.py [hunk 4]
```diff
@@ -96,6 +96,21 @@ def get_geometry(get_parameters):
...
+    Parameters:
+        rectangle: Rectangle generated from the client query.
+    Returns:
+        A 'best scale' based on the rectangle values.
+    """
+    # Get the rectangle bounds
+    x_0, y_0 = rectangle.toGeoJSON()['coordinates'][0][0]
+    x_1, y_1 = rectangle.toGeoJSON()['coordinates'][0][2]
+
+    return int((abs(x_0 - x_1) + abs(y_0 - y_1)) * 80)
```
Comment (ArchangelX360) [inline computation/imagefetcher/imagefetcher/utils.py]: Could you comment this calculation? It is just to get an order of magnitude I guess, isn't it?

File: computation/imagefetcher/imagefetcher/fetcher.py [hunk 5]
```diff
@@ -195,25 +195,58 @@ def VerticesToGeometry(vertices):
...
+            raise Error('Unsupported polygon type: %s' % geo_json['type'])
+
+        # At this point, all geo JSON are of type MultiPolygon. Since some
+        # requests may contain multiple points on the earth (such as France's
+        # DOM-TOM), we cannot generate a rectangle containing all this point
+        # (the Earth Engine API will not appreciate).
+        # We simply get the largest rectangle generated from the polygons. This
+        # may not be accurate, but at least it works!
+        def distance(x_min, y_min, x_max, y_max):
+            """Hamilton distance within two 2D points."""
```
Comment (ArchangelX360) [inline computation/imagefetcher/imagefetcher/fetcher.py]: Looks like it's a Manhattan distance (Minkowski distance with p=1), I think Hamilton distance doesn't exist. You might have drawn to much Hamiltonian paths x).

**Comment by FunkySayu** [comment]
All fixed.

File: computation/imagefetcher/imagefetcher/app.py [hunk 6]
```diff
@@ -154,6 +160,10 @@ def forest_diff_handler(polygon, place, country, city, start, stop, scale):
         'city': (city, fetcher.CityToGeometry),
     })
 
+    rectangle = fetcher.GeometryToRectangle(geometry)
+    if scale is None:
+        scale = scale_from_geometry(rectangle)
+
     current_year = date.today().year
     try:
         assert 2000 <= start < current_year, ("Start year must be within 2000 "
```
Comment (cassand) [inline computation/imagefetcher/imagefetcher/app.py:169]: Could we also specify that the date should not be the current year in the error message, as this one might be displayed in the web interface and it is likely that someone use 2017 in the search query

**Pull request merged**